src/app/tvshows/store/tvshows.actions.ts

```diff
@@ -0,0 +1,58 @@
+import { Action } from 'redux';
+import { TvShowData } from './tvshows.models';
+import { TvDbService } from '../../services';
+
+/**
+ * Actions key as enum types
+ *
+ * Typescripts's enum is ideal for this purpose, vs constants.  Plus
+ * we can use each enum value as a type specifier.  Combined with 'readonly' ensures
+ * actions keys are immutable and contain the intended value.
+ */
+
+export enum TypeKeys {
+  SEARCH_SHOWS = 'tvshows/search',
+  SEARCH_SHOWS_SUCCESS = 'tvshows/search success',
+  SEARCH_SHOWS_FAILURE = 'tvshows/search failure',
+}
+
+export interface TvShowSearchSuccess extends Action {
+  readonly type: TypeKeys.SEARCH_SHOWS_SUCCESS;
+  by: TvShowData[];
```

Comment:
Considering calling this attribute `payload`. Check out https://github.com/acdlite/flux-standard-action as a way to standardize the structure of all actions.

Reply:
Done.